libcaf_core/caf/detail/parser/read_uri.hpp

```diff
@@ -116,6 +117,19 @@ void read_uri(State& ps, Consumer&& consumer) {
   auto set_path = [&] { consumer.path(take_str()); };
   auto set_host = [&] { consumer.host(take_str()); };
   auto set_userinfo = [&] { consumer.userinfo(take_str()); };
+  auto set_colon_position = [&] {
+    colon_position = str.size();
+    str.push_back(':');
+  };
+  auto set_host_and_port = [&]() -> bool {
+    auto str = take_str();
+    if (parse(std::string_view{str}.substr(colon_position + 1), port))
```

Comment:
Instead of just picking the overflow error, it's probably better to construct a string parser state and call the other overload. Then forward the proper error code in case parsing fails.